src/infra/ssl/sslCmdRepo.go

```diff
@@ -173,75 +180,119 @@ func (repo *SslCmdRepo) shouldIncludeWww(vhost valueObject.Fqdn) bool {
 	return false
 }
 
-func (repo *SslCmdRepo) ReplaceWithValidSsl(sslPair entity.SslPair) error {
-	path, _ := valueObject.NewMappingPath(infraData.GlobalConfigs.DomainOwnershipValidationUrlPath)
-	matchPattern, _ := valueObject.NewMappingMatchPattern("equals")
-	targetType, _ := valueObject.NewMappingTargetType("inline-html")
-	httpResponseCode, _ := valueObject.NewHttpResponseCode(200)
+func (repo *SslCmdRepo) filterDomainsMappedToServer(
+	vhostNames []valueObject.Fqdn,
+	expectedOwnershipHash valueObject.Hash,
+) []valueObject.Fqdn {
+	domainsMappedToServer := []valueObject.Fqdn{}
 
-	expectedOwnershipHash, err := repo.sslQueryRepo.GetOwnershipValidationHash(
-		sslPair.Certificate.CertificateContent,
-	)
-	if err != nil {
-		return errors.New("CreateOwnershipValidationHashError: " + err.Error())
-	}
-	targetValue, _ := valueObject.NewMappingTargetValue(
-		expectedOwnershipHash.String(), targetType,
-	)
+	for _, vhostName := range vhostNames {
+		vhostNameStr := vhostName.String()
+		vhostNameIps, err := net.LookupIP(vhostNameStr)
+		if err != nil || len(vhostNameIps) == 0 {
+			continue
+		}
 
-	firstVhostName := sslPair.VirtualHostsHostnames[0]
-	inlineHtmlMapping := dto.NewCreateMapping(
-		firstVhostName,
-		path,
-		matchPattern,
-		targetType,
-		&targetValue,
-		&httpResponseCode,
-	)
+		var serverIpAddress *valueObject.IpAddress
+		for _, vhostNameIp := range vhostNameIps {
+			ipAddress, err := valueObject.NewIpAddress(vhostNameIp.String())
+			if err != nil {
+				continue
+			}
 
-	mappingCmdRepo := mappingInfra.NewMappingCmdRepo(repo.persistentDbSvc)
-	mappingId, err := mappingCmdRepo.Create(inlineHtmlMapping)
-	if err != nil {
-		return errors.New("CreateOwnershipValidationMappingError: " + err.Error())
-	}
+			serverIpAddress = &ipAddress
+			break
+		}
 
-	isDomainMappedToServer := repo.isDomainMappedToServer(
-		firstVhostName,
-		expectedOwnershipHash,
-	)
+		if serverIpAddress == nil {
+			continue
+		}
 
-	mappingQueryRepo := mappingInfra.NewMappingQueryRepo(repo.persistentDbSvc)
-	mappings, err := mappingQueryRepo.ReadByHostname(firstVhostName)
-	if err != nil {
-		return errors.New("ReadVhostMappingsError: " + err.Error())
+		ownershipValidationMappingId, err := repo.createOwnershipValidationMapping(
```

Comment:
Ok, you're checking via DNS and via HTTP. So, that's wrong for sure. You should have checked the DNS on the first method to not only eliminate those not pointing to "somewhere" but also eliminate those not pointing to the server itself since you'll be ignoring these domains anyway.